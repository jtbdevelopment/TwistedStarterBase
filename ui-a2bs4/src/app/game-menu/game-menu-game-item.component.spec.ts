import {TestBed} from '@angular/core/testing';
import {RouterTestingModule} from '@angular/router/testing';
import {GameMenuGameItemComponent} from './game-menu-game-item.component';
import {Game, MapKeysPipe} from 'jtb-core-games-ui';


describe('Component:  game menu game item component', () => {
    beforeEach(() => {
        TestBed.configureTestingModule({
            imports: [
                RouterTestingModule
            ],
            declarations: [
                GameMenuGameItemComponent,
                MapKeysPipe
            ],
        });
        TestBed.compileComponents();
    });

    it('displays game', () => {
        const fixture = TestBed.createComponent(GameMenuGameItemComponent);
        fixture.componentInstance.game = new Game({
            id: '1234',
            gamePhase: 'Playing',
            players: {'md51': 'p1', 'md52': 'p2'},
            playerImages: {'md51': '', 'md52': ''}
        });
        fixture.detectChanges();
        expect(fixture).toMatchSnapshot();
    });

    it('default describe game function just returns todo', () => {
        const fixture = TestBed.createComponent(GameMenuGameItemComponent);
        expect(fixture.componentInstance.describeGame()).toEqual('TODO');
    });

});
