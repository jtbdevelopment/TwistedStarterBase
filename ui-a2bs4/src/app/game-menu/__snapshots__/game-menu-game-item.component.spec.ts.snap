// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component:  game menu game item component displays game 1`] = `
<app-game-menu-game-item
  game={[Function Game]}
>
  <a
    class="btn-group btn btn-light btn-group-game btn-group-game-"
    href="/game/playing/1234"
    ng-reflect-router-link="/game/playing/1234"
  >
    
      
    
    <img
      class="player-image"
      src=""
    />
    <img
      class="player-image"
      src=""
    />
    
      
    <span
      class="game-description"
    >
      TODO
    </span>
    
  
  </a>
  
  
</app-game-menu-game-item>
`;
